libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -147,6 +151,22 @@ class MapboxNavigation(
         tripSession.registerStateObserver(navigationSession)
     }
 
+    /**
+     * Tell the on-board navigator to start loading routing tiles. This feature is in development
+     * and does not give feedback on the status of the tiles being loaded, and does not manage
+     * stale router configurations that were loaded. This is not ready for production use.
+     *
+     * @param routerTilesParams describes the version of the router tiles
+     * @see [NavigationRouterParamsProvider] gives default values for the RouterTilesParams
+     */
+    fun configureRouter(routerTilesParams: RouterTilesParams) {
```

Comment:
Do we need to set a different tile endpoints while `MapboxNaivgation` is alive? Wouldn't `MapboxOnboardRouterConfig` that's passed with `NavigationOptions` be enough?

Reply:
Yeah, that's a good point. We need to think about this because we may also need to configure the router for _Offboard_ and _Hybrid_ in order to get the NN routing tiles pre-fetching.

Reply:
Are we creating multiple navigators inside? Need to be careful with this as it impacts the caching.

Reply:
Nah, we're 🆗 on that side, `MapboxNativeNavigatorImpl` is a singleton 👀 https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigator/src/main/java/com/mapbox/navigation/navigator/MapboxNativeNavigatorImpl.kt#L34 Thanks for the heads up @mskurydin 🙇

Reply:
So it looks like we should be using this to send the router tile endpoint

https://github.com/mapbox/mapbox-navigation-android/blob/83da15a0052645546da07b26beef9e8eaa1ec229/libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt#L64-L78

Reply:
@Guardiola31337 except will this work? we're creating a File here. But the RouterParams are only a path. 

Is NN able to deal with Android file management? Seems like no?

Reply:
OK this will put the file management responsibility on the app that integrates it for now. That's probably better for now. It would be nice that have a default impl available in the SDK eventually so it's easy for those who integrate it

Reply:
@kmadsen When you have a chance could you cut a ticket around ☝️ so we don't forget to look into this? 🙏

Reply:
This seems like a MVP requirement for offline routing right? Looks like there has been some work by @RingerJK here

https://github.com/mapbox/navigation-sdks/issues/195
https://github.com/mapbox/mapbox-navigation-android/blob/master/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloader.kt

It seems like we haven't decided where to put this file management responsibility. This seems like something that the SDK should do, but it appears NN is also adding the ability to download tiles?

Reply:
That's a wrapper around [Route Tiles API](https://github.com/mapbox/mapbox-java/tree/master/services-route-tiles) so clients can download routing tiles and get those available in the specified path (Route Tiles API returns a zip file that needs to be uncompressed following a specific structure), which is unrelated from NN downloading tiles. NN though is exposing a callback (`HttpInterface`) so that clients can do the request on their side (so they can use the HTTP client of preference and NN doesn't need to integrate HTTP support on their side which is not optimal to the platform clients).

Please cut a separate ticket and we can discuss further.